### PR TITLE
Implemented asset id converter for xtoken pallet

### PR DESCRIPTION
**Pull request by zktony**
Implemented asset id converter for xtoken pallet

### CODE REVIEW

File: pallets/xcm-handler/src/lib.rs [hunk 1]
```diff
@@ -598,5 +611,20 @@ pub mod pallet {
...
+
+		pub fn convert_asset_id_to_location(asset_id: u128) -> Option<MultiLocation> {
+			let (_, _, asset_identifier) = <TheaAssets<T>>::get(asset_id);
+			let asset_identifier = asset_identifier.to_vec();
+			let parachain_asset: Option<ParachainAsset> = Decode::decode(&mut &asset_identifier[..]).ok();
+			if let Some(asset) = parachain_asset {
+				Some(asset.location)
+			} else {
+				None
+			}
```
Comment (serhii-temchenko) [inline pallets/xcm-handler/src/lib.rs:623]: I suggest to change this to something like:

```rust
Decode::decode(&mut &asset_identifier[..]).ok().map(|asset: ParachainAsset| asset.location)

```

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/xcm-handler/src/lib.rs (hunk 1 above)
Comment (zktony) [reply pallets/xcm-handler/src/lib.rs]: done

**Review by serhii-temchenko — APPROVED** [review]

File: pallets/xcm-handler/src/lib.rs [hunk 2]
```diff
@@ -352,6 +353,18 @@ pub mod pallet {
...
 
+    impl<T: Config>	Convert<u128, Option<MultiLocation>> for Pallet<T> {
+		fn convert(asset_id: u128) -> Option<MultiLocation> {
+            Self::convert_asset_id_to_location(asset_id)
+		}
+	}
+
+	impl<T: Config>	Convert<MultiLocation, Option<u128>> for Pallet<T> {
+		fn convert(a: MultiLocation) -> Option<u128> {
+			todo!()
```
Comment (Gauthamastro) [inline pallets/xcm-handler/src/lib.rs:364]: Fix this.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/xcm-handler/src/lib.rs (hunk 2 above)
Comment (zktony) [reply pallets/xcm-handler/src/lib.rs]: done

File: pallets/xcm-handler/src/lib.rs (hunk 1 above)
Comment (serhii-temchenko) [reply pallets/xcm-handler/src/lib.rs]: I assume after resolving conflicts you moved back the old implementation. Please update it again.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]
Please check [this](https://github.com/Polkadex-Substrate/parachain/pull/14#discussion_r1112257693) comment.

File: pallets/xcm-handler/src/lib.rs (hunk 1 above)
Comment (zktony) [reply pallets/xcm-handler/src/lib.rs]: oops .. I added it back

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**